### DISCUSSION TITLE
Connecting multiple temperature sensors and drawing curves

### DISCUSSION THREAD

**Discussion by Moro2202** [artisan-roaster-scope/artisan#832] [Q&A] [answered]
I want to connect 8 temperature sensors with 4 Arduino boards (2 sensors are connected to one board). The communication protocol is MODBUS. I want the readings of all sensors to be displayed in the window as 8 curves.
Can this be done in Artisan with the current functionality? Or do I need to modify Artisan?

**Comment by MAKOMO** [comment, marked as answer]
Artisan can fetch data from up to 8 registers also if hosted on different slaves. All slaves need to be connected to the same RS485 MODBUS ring in case of MODBUS RTU/ASCII or accessible via the same IP in case of MODBUS TCP/UDP.

**Comment by Moro2202** [comment]
Thanks for the clarification.
Is it possible to display 8 curves on the graph?

**Comment by MAKOMO** [comment]
Any number. Note that you can add more curves as Virtual Device curves with data computed via Symbolic Formula from the existing data.
